activerecord/test/cases/explain_test.rb

```diff
@@ -89,22 +89,26 @@ def test_relation_explain_with_sum
       assert_match(expected_query, message)
     end
 
-    def test_relation_explain_with_first
-      expected_query = capture_sql {
-        Car.all.first
-      }.first
-      message = Car.all.explain.first
-      assert_match(/^EXPLAIN/, message)
-      assert_match(expected_query, message)
-    end
+    unless current_adapter?(:Mysql2Adapter) && ActiveRecord::Base.lease_connection.prepared_statements
```

Comment:
Is it valid to lease the connection here?

I think we can just change this class to override `capture_sql` as `unprepared_statement { super }` (for MySQL)

Reply:
The entire file is under `if ActiveRecord::Base.lease_connection.supports_explain?` so I think the lease is fine. 

But I'll dig a bit deeper to see if this can be cleaned up.